File: subroutines/all/subroutine dsmatvec.f
      subroutine dsmatvec(rot,din,dout)
      real*8 din(3),dout(3),d(3)
      dimension rot(3,3)
      do k=1,3
        rr=0.0
        do i=1,3
          rr=rr+din(i)*rot(k,i)
        end do
        d(k)=rr
      end do
      call trnsfrd(dout,d,3)
      return
      end

File: python translations/subroutine dsmatvec.py
import numpy as np

def dsmatvec(rot, din):
	"""
	Compute dout = din*rot (Vector*Matrix)
	***
	Parameters:
	rot: rotational matrix 3*3
	din: input vector 
	"""
	dout = np.zeros(3)
	for k in range(3):
		dout[k]=sum([din[i]*rot[k][i] for i in range(3)])
	return dout
